pkg/providers/aws/cluster_network_provider.go

```diff
@@ -26,77 +38,620 @@ import (
-	return errorUtil.New("CreateNetwork stub")
-}
+	clusterID, err := resources.GetClusterID(ctx, n.Client)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error getting clusterID")
+	}
+	organizationTag := resources.GetOrganizationTag()
 
-func (n *NetworkProvider) DeleteNetwork() error {
-	logger := n.logger.WithField("action", "DeleteNetwork")
-	logger.Debug("DeleteNetwork stub")
-	return nil
+	// check if there is cluster specific vpc already created.
+	foundVpc, err := getStandaloneVpc(n.Ec2Api, logger, clusterID, organizationTag)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "unable to get vpc")
+	}
+	if foundVpc == nil {
+		//VPCs must be created with a valid CIDR Block range, between \16 and \26
+		//if an invalid range is passed, the function returns an error
+		//
+		//VPCs are tagged with the name `<organizationTag>/clusterID`.
+		//By default, `integreatly.org/clusterID`.
+		//
+		//NOTE - Once a VPC is created we do not want to update it. To avoid changing cidr block
+		if !isValidCIDRRange(vpcCidrBlock) {
+			return nil, errorUtil.New(fmt.Sprintf("%s is out of range, block sizes must be between `/16` and `/26`, please update `_network` strategy", vpcCidrBlock.String()))
+		}
+
+		logger.Infof("cidr %s is valid 👍", vpcCidrBlock.String())
+
+		// create vpc using cidr string from _network
+		createVpcOutput, err := n.Ec2Api.CreateVpc(&ec2.CreateVpcInput{
+			CidrBlock: aws.String(vpcCidrBlock.String()),
+		})
+		ec2Err, isAwsErr := err.(awserr.Error)
+		if err != nil && isAwsErr && ec2Err.Code() == "InvalidVpc.Range" {
+			return nil, errorUtil.New(fmt.Sprintf("%s is out of range, block sizes must be between `/16` and `/26`, please update `_network` strategy", vpcCidrBlock.String()))
+		}
+		if err != nil {
+			return nil, errorUtil.Wrap(err, "unexpected error creating vpc")
+		}
+		logger.Infof("creating vpc: %s for clusterID: %s", *createVpcOutput.Vpc.VpcId, clusterID)
+
+		// ensure standalone vpc has correct tags
+		clusterIDTag := &ec2.Tag{
+			Key:   aws.String(fmt.Sprintf("%sclusterID", organizationTag)),
+			Value: aws.String(clusterID),
+		}
+		if err = n.reconcileVPCTags(createVpcOutput.Vpc, clusterIDTag); err != nil {
+			return nil, errorUtil.Wrapf(err, "unexpected error while reconciling vpc tags")
+		}
+
+		return &Network{
+			Vpc: createVpcOutput.Vpc,
+		}, nil
+	}
+
+	// reconciling on vpc networking, ensuring the following are present :
+	//     * subnets (2 private)
+	//     * subnet groups -> rds and elasticache
+	//     * TODO security groups - https://issues.redhat.com/browse/INTLY-8105
+	//     * TODO route table configuration - https://issues.redhat.com/browse/INTLY-8106
+	//
+	privateSubnets, err := n.reconcileStandaloneVPCSubnets(ctx, n.Logger, foundVpc, clusterID, organizationTag)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "unexpected error creating vpc subnets")
+	}
+
+	// create rds subnet group
+	if err = n.reconcileStandaloneRDSVpc(ctx, privateSubnets, clusterID); err != nil {
```

Comment:
I feel like these functions are named incorrectly. It reads like there's a separate VPC for RDS and Elasticache.

Maybe `reconcileRDSVPCConfiguration` or something like that?